vm/array.go

```diff
@@ -90,533 +81,547 @@ func (a *ArrayObject) shift() Object {
 	return value
 }
 
-// initArrayObject returns an array that contains given objects
-func initArrayObject(elements []Object) *ArrayObject {
-	return &ArrayObject{Elements: elements, Class: arrayClass}
+func (vm *VM) initArrayObject(elements []Object) *ArrayObject {
+	return &ArrayObject{Elements: elements, Class: vm.builtInClasses[arrayClass]}
 }
 
-func initArrayClass() {
-	bc := &BaseClass{Name: "Array", ClassMethods: newEnvironment(), Methods: newEnvironment(), Class: classClass, pseudoSuperClass: objectClass, superClass: objectClass}
-	ac := &RArray{BaseClass: bc}
-	ac.setBuiltInMethods(builtinArrayInstanceMethods, false)
-	arrayClass = ac
+func (vm *VM) initArrayClass() *RClass {
+	ac := vm.initializeClass(arrayClass, false)
+	ac.setBuiltInMethods(builtinArrayInstanceMethods(), false)
+	ac.setBuiltInMethods(builtInArrayClassMethods(), true)
+	return ac
 }
 
-var builtinArrayInstanceMethods = []*BuiltInMethodObject{
-	{
-		// Retrieves an object in an array using Integer index.
-		// The index starts from 0. It returns `null` if the given index is bigger than its size.
-		//
-		// ```ruby
-		// a = [1, 2, 3, "a", "b", "c"]
-		// a[0]  # => 1
-		// a[3]  # => "a"
-		// a[10] # => nil
-		// a[-1] # => "c"
-		// a[-3] # => "a"
-		// a[-7] # => nil
-		// ```
-		Name: "[]",
-		Fn: func(receiver Object) builtinMethodBody {
-			return func(t *thread, args []Object, blockFrame *callFrame) Object {
-				if len(args) != 1 {
-					return &Error{Message: "Expect 1 arguments. got=%d" + string(len(args))}
+func builtInArrayClassMethods() []*BuiltInMethodObject {
```

Comment:
@adlerhsieh Now all methods are stored in functions (both built in classes and standard libs), so you might need to change the API document parser to adjust this.